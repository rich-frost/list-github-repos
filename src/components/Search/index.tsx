import { Flex, Text, Input, Center } from '@chakra-ui/react';

interface SearchProps {
  search: string;
  setSearch: React.Dispatch<React.SetStateAction<string>>;
}
export default function Search({ search, setSearch }: SearchProps) {
  return (
    <Flex justify="space-evenly">
      <Center w="200px">
        <Text>Search repos by topic</Text>
      </Center>

      <Input
        placeholder="Enter topic"
        onChange={(e) => setSearch(e.target.value)}
        value={search}
      />
    </Flex>
  );
}
