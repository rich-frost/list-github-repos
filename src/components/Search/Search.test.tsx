import renderer from 'react-test-renderer';
import Search from './index';
import { render, fireEvent } from '@testing-library/react';

describe('Search component', () => {
  it('renders the component', () => {
    const setSearch = jest.fn();
    const component = renderer.create(
      <Search search="" setSearch={setSearch} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the component with populated input', () => {
    const setSearch = jest.fn();
    const component = renderer.create(
      <Search search="" setSearch={setSearch} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls setSearch callback on input change', async () => {
    const searchParam = 'golang';
    const setSearch = jest.fn();
    const { container } = render(<Search search="" setSearch={setSearch} />);

    const input = await container.getElementsByClassName('chakra-input')[0];

    fireEvent.change(input, { target: { value: searchParam } });

    expect(setSearch).toHaveBeenCalledTimes(1);
    expect(setSearch).toHaveBeenCalledWith(searchParam);
  });
});
