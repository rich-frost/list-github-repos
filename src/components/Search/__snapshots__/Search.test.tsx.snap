// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders the component 1`] = `
<div
  className="css-106l0id"
>
  <div
    className="css-9ck52x"
  >
    <p
      className="chakra-text css-0"
    >
      Search repos by topic
    </p>
  </div>
  <input
    className="chakra-input css-0"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter topic"
    value=""
  />
</div>
`;

exports[`Search component renders the component with populated input 1`] = `
<div
  className="css-106l0id"
>
  <div
    className="css-9ck52x"
  >
    <p
      className="chakra-text css-0"
    >
      Search repos by topic
    </p>
  </div>
  <input
    className="chakra-input css-0"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter topic"
    value=""
  />
</div>
`;
